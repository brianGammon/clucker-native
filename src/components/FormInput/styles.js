import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  label: {
    fontWeight: 'bold',
  },
  revealPassword: {
    marginLeft: 5,
    marginRight: 5,
    color: 'grey',
    fontSize: 32,
  },
  errorContainer: {
    height: 20,
  },
  error: {
    color: 'red',
    paddingTop: 2,
    paddingLeft: 4,
    fontSize: 14,
  },
});
