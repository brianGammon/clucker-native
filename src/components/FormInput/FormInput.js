/* @flow */
import React from 'react';
import {
  View, Input, Item, Icon, Text, Label, Button,
} from 'native-base';
import errorMapper from '../../utils/errorMapper';
import styles from './styles';

type Props = {
  handler: () => void,
  touched: boolean,
  errors: {
    [errorKey: string]: any,
  } | null,
  meta: {
    autoCapitalize: 'none' | 'sentences' | 'words' | 'characters',
    label: string,
    secureTextEntry: boolean,
    multiline: boolean,
    maxLength: number,
    numberOfLines: number,
    keyboardType: any,
  },
};

type State = {
  hidePassword: boolean,
};

class FormInput extends React.Component<Props, State> {
  state = { hidePassword: true };

  toggleMask = (hidePassword: boolean) => {
    this.setState({ hidePassword });
  };

  render() {
    const {
      handler, touched, meta, errors,
    } = this.props;
    const { hidePassword } = this.state;
    const error = errorMapper(errors, meta);
    return (
      <View style={styles.container}>
        <Item inlineLabel error={touched && error !== null}>
          <Label style={styles.label}>{meta.label}:</Label>
          <Input
            keyboardType={meta.keyboardType || null}
            maxLength={meta.maxLength || null}
            numberOfLines={meta.numberOfLines || null}
            multiline={meta.multiline}
            // placeholder={meta.label}
            autoCapitalize={meta.autoCapitalize}
            secureTextEntry={hidePassword && meta.secureTextEntry}
            {...handler()}
          />
          {touched && error !== null && <Icon name="alert" />}
          {meta.secureTextEntry && (
            <Button transparent onPress={() => this.toggleMask(!hidePassword)}>
              <Icon
                style={styles.revealPassword}
                name={hidePassword ? 'eye' : 'eye-off'}
              />
            </Button>
          )}
        </Item>
        <View style={styles.errorContainer}>
          <Text style={styles.error}>{touched && error}</Text>
        </View>
      </View>
    );
  }
}

export default FormInput;
