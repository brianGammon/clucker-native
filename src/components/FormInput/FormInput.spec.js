import React from 'react';
import FormInput from '.';

const props = {
  handler: () => {},
  touched: false,
  errors: {
    required: true,
    email: true,
  },
  meta: {
    autoCapitalize: 'none',
    mustMatchLabel: 'otherField',
    label: 'thisField',
    secureTextEntry: true,
  },
};

test('FormInput renders - error, not touched yet', () => {
  const wrapper = shallow(<FormInput {...props} />);
  expect(wrapper).toMatchSnapshot();
});

test('FormInput renders - errors, touched', () => {
  const theseProps = { ...props, touched: true };
  const wrapper = shallow(<FormInput {...theseProps} />);
  expect(wrapper).toMatchSnapshot();
});
