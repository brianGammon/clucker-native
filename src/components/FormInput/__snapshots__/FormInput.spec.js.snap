// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInput renders - error, not touched yet 1`] = `
<Styled(ViewNB)>
  <Styled(Item)
    error={false}
    inlineLabel={true}
  >
    <Styled(Label)
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      thisField
      :
    </Styled(Label)>
    <Styled(Input)
      autoCapitalize="none"
      keyboardType={null}
      maxLength={null}
      numberOfLines={null}
      secureTextEntry={true}
    />
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="eye"
        style={
          Object {
            "color": "grey",
            "fontSize": 32,
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      />
    </Styled(Button)>
  </Styled(Item)>
  <Styled(ViewNB)>
    <Styled(Text)
      style={
        Object {
          "color": "red",
          "fontSize": 14,
          "paddingLeft": 4,
          "paddingTop": 2,
        }
      }
    />
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;

exports[`FormInput renders - errors, touched 1`] = `
<Styled(ViewNB)>
  <Styled(Item)
    error={true}
    inlineLabel={true}
  >
    <Styled(Label)
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      thisField
      :
    </Styled(Label)>
    <Styled(Input)
      autoCapitalize="none"
      keyboardType={null}
      maxLength={null}
      numberOfLines={null}
      secureTextEntry={true}
    />
    <Styled(Icon)
      name="alert"
    />
    <Styled(Button)
      onPress={[Function]}
      transparent={true}
    >
      <Styled(Icon)
        name="eye"
        style={
          Object {
            "color": "grey",
            "fontSize": 32,
            "marginLeft": 5,
            "marginRight": 5,
          }
        }
      />
    </Styled(Button)>
  </Styled(Item)>
  <Styled(ViewNB)>
    <Styled(Text)
      style={
        Object {
          "color": "red",
          "fontSize": 14,
          "paddingLeft": 4,
          "paddingTop": 2,
        }
      }
    >
      Enter a valid email address.
    </Styled(Text)>
  </Styled(ViewNB)>
</Styled(ViewNB)>
`;
